### PR TITLE
Disable Auto-update

### PR DESCRIPTION
**Pull request by cRui861**
Setting a command line flag to disable auto-update upon application launch.

### CODE REVIEW

**Comment by timotheeguerin** [comment]
Why is this needed? There is a settings that disable auto update on quit already.

File: src/client/cli.ts [hunk 1]
```diff
@@ -11,6 +12,7 @@ export interface BatchExplorerArgs {
 export function parseArguments(argv: string[]): BatchExplorerArgs {
     return commander
         .version(app.getVersion())
+        .option("--do-not-autoupdate", "Will not autoupdate the application")
```
Comment (bgklein) [inline src/client/cli.ts]: should likely make this --autoupdate <true|false> so the user can either enable or disable

File: src/client/startup.ts [hunk 2]
```diff
@@ -68,6 +68,11 @@ export async function startBatchExplorer(args: BatchExplorerArgs) {
         app.commandLine.appendSwitch("ignore-certificate-errors", "true");
     }
 
+    if (args.doNotAutoupdate) {
+        log.warn("Application will not autoupdate");
+        autoUpdater.autoInstallOnAppQuit = false;
```
Comment (bgklein) [inline src/client/startup.ts]: Should likelymake this both set this and update config

File: src/client/cli.ts (hunk 1 above)
Comment (xingwu1) [reply src/client/cli.ts]: agree. Just use autoupdate, and set default to false.

Comment (cRui861) [reply src/client/cli.ts]: If the flag is set, that means that value is already set to true so in this case, that means autoupdate will be turned off. When the user does not set a flag, then the value is false so in this case, the user did not try to turn off autoupdate.

**Comment by timotheeguerin** [comment]
A little confused by the point of having this flag. Do you expect user to start BatchExplorer from the command line with the flag just to not be able to auto update?

File: src/client/cli.ts (hunk 1 above)
Comment (cRui861) [reply src/client/cli.ts]: Discussed offline

File: src/client/cli.ts [hunk 3]
```diff
@@ -2,15 +2,19 @@ import * as commander from "commander";
...
     ignoreCertificateErrors?: boolean;
     args: string[];
 }
 
+// add user config options (in Settings) to also let the status of the autoupdate to persist - whether that be on or off
+// change naming to "set autoupdate on/off" or "autoupdate set to true/false" -> as Xing/Matt
 export function parseArguments(argv: string[]): BatchExplorerArgs {
     return commander
         .version(app.getVersion())
+        .option("--disable-autoupdate", "Disables autoupdate for instance and changes settings to turn off autoupdate")
```
Comment (bgklein) [inline src/client/cli.ts]: auto-update is two words so add dashes

**Comment by cRui861** [comment]
@timotheeguerin Yeah, the goal is to allow users to use this flag to disable auto-update when opening the app for an instance. I've noted your other suggestions mentioned offline and will revisit them.

**Comment by cRui861** [comment]
Fixes #2152

**Pull request merged**